server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -396,8 +367,32 @@ void executeRequest(
                     searchPhaseProvider.apply(delegate)
                 );
             } else {
+                if ((listener instanceof TelemetryListener tl) && CCS_TELEMETRY_FEATURE_FLAG.isEnabled()) {
+                    tl.setRemotes(resolvedIndices.getRemoteClusterIndices().size());
+                    if (isAsyncSearchTask(task)) {
+                        tl.setFeature(CCSUsageTelemetry.ASYNC_FEATURE);
+                    }
+                    String client = task.getHeader(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER);
+                    if (client != null) {
+                        tl.setClient(client);
+                    }
+                    // Check if any of the index patterns are wildcard patterns
+                    var localIndices = resolvedIndices.getLocalIndices();
+                    if (localIndices != null && Arrays.stream(localIndices.indices()).anyMatch(Regex::isSimpleMatchPattern)) {
+                        tl.setFeature(CCSUsageTelemetry.WILDCARD_FEATURE);
+                    }
+                    if (resolvedIndices.getRemoteClusterIndices()
+                        .values()
+                        .stream()
+                        .anyMatch(originalIndices -> Arrays.stream(originalIndices.indices()).anyMatch(Regex::isSimpleMatchPattern))) {
```

Comment:
This stream construct looks funny to me. Why are you calling anyMatch twice? The first one seems wrong?

Reply:
It's twice because it is looking for any element of originalIndices that has any index that contains a wildcard.